src/pretix/base/templatetags/rich_text.py

```diff
@@ -144,20 +159,32 @@ def truelink_callback(attrs, new=False):
 
         <a href="https://maps.google.com/location/foo">https://maps.google.com</a>
     """
-    text = re.sub(r'[^a-zA-Z0-9.\-/_ ]', '', attrs.get('_text'))  # clean up link text
+    text = re.sub(r'[^a-zA-Z0-9.\-/_@: ]', '', attrs.get('_text'))  # clean up link text
     url = attrs.get((None, 'href'), '/')
     href_url = urllib.parse.urlparse(url)
+
+    # Verify server name of URL names
     if (None, 'href') in attrs and URL_RE.match(text) and href_url.scheme not in ('tel', 'mailto'):
         # link text looks like a url
         if text.startswith('//'):
             text = 'https:' + text
         elif not text.startswith('http'):
             text = 'https://' + text
 
+        text_url = urllib.parse.urlparse(text)
+        if text_url.netloc.split("@")[-1] != href_url.netloc.split("@")[-1] or not href_url.path.startswith(text_url.path):
+            # link text contains an URL that has a different base than the actual URL
+            attrs['_text'] = attrs[None, 'href']
+
+    # Verify server name of mastodon display names (@name@server.tld)
+    if (None, 'href') in attrs and FEDIVERSE_RE.match(text):
+        parts = text.split('@')
```

Comment:
Not a huge deal, but I can create a markdown-link to another user like so:

`[@good@mastodon.social](https://mastodon.social/@bad)`

Is that something we want to allow?